src/features/hostapp/hooks/target-hostapp.ts

```diff
@@ -55,7 +58,46 @@ hooks.addPureHook('PATCH', 'resin', 'device', {
 			);
 		}
 	},
-});
+};
+
+const postHostappHook: hooks.Hooks = {
+	async POSTPARSE(args) {
+		if (
+			args.request.values.os_version != null &&
+			args.request.values.os_variant != null
+		) {
+			let deviceType;
+			if (!args.request.values.is_of__device_type) {
+				[deviceType] = await args.api.get({
+					resource: 'device_type',
+					options: {
+						$select: 'id',
+						$filter: {
+							slug: { $eq: args.request.values.device_type },
```

Comment:
`$eq` is normally unnecessary:
```suggestion
							slug: args.request.values.device_type,
```